### PR TITLE
Add data

### PR DESCRIPTION
**Pull request by avkoll**
I added more data in the output and cleaned up some of the code in the forecast functions. 

Data added:
- UV Index
- Air temp min and max
- Rain amount in inches
- Max precipitation chance
- Max wind speed
- Main wind direction

So far these are only as specific as max or min each day, does not address hourly temp changes.

No units are printed right now.

I didn't want to change too much all around so I just returned each item in a dictionary in the order that they are printed so I wouldn't have to rewrite the print_forecast function in helper.py.

This is my first contribution, please let me know if it doesn't follow the styleguide.
Addresses issue #4 More Data

## Summary by Sourcery

Enhance the forecast output by adding additional weather data metrics including UV Index, air temperature (min and max), rain amount, max precipitation chance, max wind speed, and main wind direction. Refactor the forecast function to streamline data extraction using loops.

New Features:
- Add UV Index, air temperature (min and max), rain amount, max precipitation chance, max wind speed, and main wind direction to the forecast output.

Enhancements:
- Refactor forecast function to use loops for extracting marine and general weather data, reducing the number of local variables.

### CODE REVIEW

**Comment by avkoll** [comment]
@sourcery-ai review

**Comment by ryansurf** [comment]
Hey @avkoll, these changes look great! I'll get to reviewing this tomorrow AM, but all checks have passed so its looking good 👍

**Comment by avkoll** [comment]
@ryansurf Great! Sorry I saw someone else wanted to work on the same issue, I'll claim an issue next time before I start woring on it.

**Comment by ryansurf** [comment]
> @ryansurf Great! Sorry I saw someone else wanted to work on the same issue, I'll claim an issue next time before I start woring on it.

No worries, that was my fault. Good work

File: src/api.py [hunk 1]
```diff
@@ -177,27 +177,48 @@ def forecast(lat, long, decimal, days=0):
...
+        "wind_speed_unit": "mph",
+        "precipitation_unit": "inch",
+        "timezone": "auto",
+        "forecast_days": days,
+    }
 
-    responses = openmeteo.weather_api(urls[0], params=params)
-    responses_uv = openmeteo.weather_api(urls[1], params=params_uv)
+    responses_marine = openmeteo.weather_api(urls[0], params=params)
+    responses_general = openmeteo.weather_api(urls[1], params=params_general)
```
Comment (ryansurf) [inline src/api.py:200]: Good naming of variables

File: src/api.py [hunk 2]
```diff
@@ -177,27 +177,48 @@ def forecast(lat, long, decimal, days=0):
...
 
-    response = responses[0]
-    response_uv = responses_uv[0]
+    response = responses_marine[0]
```
Comment (ryansurf) [inline src/api.py:202]: Perhaps `response_marine` would have been a better name, for consistency

File: src/api.py [hunk 3]
```diff
@@ -177,27 +177,48 @@ def forecast(lat, long, decimal, days=0):
...
-    )
-    daily_period_max = helper.round_decimal(
-        response.Daily().Variables(2).ValuesAsNumpy(), decimal
-    )
+    # Extract marine data using a loop
+    marine_data = [
+        helper.round_decimal(
+            response.Daily().Variables(i).ValuesAsNumpy(), decimal
+        )
+        for i in range(3)
```
Comment (ryansurf) [inline src/api.py:210]: niceeee

File: src/helper.py [hunk 4]
```diff
@@ -152,6 +152,18 @@ def print_forecast(ocean, forecast):
             print("Wave Period: ", day[2])
         if int(ocean["show_uv"]) == 1:
             print("UV Index: ", day[4])
+        if int(ocean["show_air_temp"]) == 1:
+            print("Air Temp Max: ", day[5])
+        if int(ocean["show_air_temp"]) == 1:
+            print("Air Temp Min: ", day[6])
+        if int(ocean["show_air_temp"]) == 1:
+            print("Rain Sum: ", day[7])
+        if int(ocean["show_air_temp"]) == 1:
```
Comment (ryansurf) [inline src/helper.py:161]: Why do rain sum and precipitation probability show if `show_air_temp == 1`? We should create new variables, like `"show_rain_sum": 0` and let the user specify if they want to see it in a cli argument, like show_rain_sum/srs

See `set_output_values(args, arguments)` in `helper.py` for an example

Comment (ryansurf) [reply src/helper.py:161]: Also, if you make this change make sure to include it in the README's arguments section!

File: src/helper.py [hunk 5]
```diff
@@ -27,9 +27,9 @@ def arguments_dictionary(lat, long, city, args):
         "show_period": 1,
         "show_city": 1,
         "show_date": 1,
-        "show_air_temp": 0,
-        "show_wind_speed": 0,
-        "show_wind_direction": 0,
+        "show_air_temp": 1,
+        "show_wind_speed": 1,
+        "show_wind_direction": 1,
```
Comment (ryansurf) [inline src/helper.py:32]: These should all be set to 0, and only show if the user wants them to (via a command line argument, like `show_wind_speed`

**Review by ryansurf — CHANGES_REQUESTED** [review]
Hey @avkoll, nice work! I've left a couple comments suggesting some changes, but your code runs and is almost there. 

Basically, the data you including should default to being hidden and the user should specify if they want to see it in a command line argument (like: `curl localhost:8000?show_rain_sum` or `curl localhost:8000?srs`). 

You have all of your new data, like `rain_sum`, show if `show_air_temp == 1`, but they should all have their own boolean variable, like `show_rain_sum : 0` etc. 

Let me know if these comments make sense! I'm happy to help out if need be

File: src/helper.py (hunk 4 above)
Comment (avkoll) [reply src/helper.py]: I changed those while I was testing so they would show by default when I was adding them to the forecast... forgor to change them back.

**Comment by avkoll** [comment]
I think that should be it, also is it common practice to respond to each change request individually?

**Comment by avkoll** [comment]
oops

**Comment by ryansurf** [comment]
> I think that should be it, also is it common practice to respond to each change request individually?

You don't have to, but its nice to do!

**Comment by ryansurf** [comment]
So I reviewed your PR again. Works mostly great, the forecast/json shows the rain sum/precipitation probability. 

However, the current data does not shows these. When running `curl localhost:8000?srs,spp`, neither are shown! However, they are shown when a forecast is passed as an arg.

**Comment by avkoll** [comment]
Sorry I was focused on the forecast function and completely forgot about the current data. I made a new function for the daily rain data. 

Since the ocean_data_dict is getting sort of big maybe there could be dictionaries for current, daily, and hourly data?

**Comment by ryansurf** [comment]
@avkoll nice. So now the current rain data works but the JSON output is now broken :flushed:

I'm gonna merge it anyways. I have some time right now and I think the whole helper file needs a rehaul. But I agree, we can separate the dictionaries to make thinks more simple.

**Comment by ryansurf** [comment]
@avkoll nice. So now the current rain data works but the JSON output is now broken :flushed:

I'm gonna merge it anyways. I have some time right now and I think the whole helper file needs a rehaul. But I agree, we can separate the dictionaries to make thinks more simple.

**Pull request merged**